File: usecase/core/src/main/java/com/hearlers/gateway/port/AuthUserPort.java
package com.hearlers.gateway.port;

import com.hearlers.api.proto.v1.model.AuthChannel;
import com.hearlers.api.proto.v1.model.AuthUser;
import com.hearlers.api.proto.v1.service.*;

public interface AuthUserPort {
    AuthUser getAuthUser(String uniqueId, AuthChannel authChannel);
    InitializeUserResponse initializeUser(InitializeUserRequest request);
    ConnectAuthChannelResponse connectAuthChannel(ConnectAuthChannelRequest request);
    UpdateAuthorityResponse updateAuthority(UpdateAuthorityRequest request);
}

File: usecase/core/src/main/kotlin/com/hearlers/gateway/port/AuthUserPort.kt
package com.hearlers.gateway.port

import com.hearlers.api.proto.v1.model.AuthChannel
import com.hearlers.api.proto.v1.model.AuthUser
import com.hearlers.api.proto.v1.service.*

interface AuthUserPort {
    fun getAuthUser(uniqueId: String, authChannel: AuthChannel): AuthUser
    fun initializeUser(request: InitializeUserRequest): InitializeUserResponse
    fun connectAuthChannel(request: ConnectAuthChannelRequest): ConnectAuthChannelResponse
    fun updateAuthority(request: UpdateAuthorityRequest): UpdateAuthorityResponse
}
